lib/application/workflow/cli/wikipedia_handler.ex

```diff
@@ -0,0 +1,36 @@
+defmodule WikipediaHandler do
+  def handle_wikipedia_article(uri) do
+    main_article = fetch_wikipedia_page(uri, 3)
+    references = fetch_references(main_article.references)
+    IO.puts("fetched refs: #{Enum.count(references)}")
+  end
+
+  defp fetch_wikipedia_page(uri, attempts) do
+    case Wikipedia.fetch_content(uri) do
+      {:ok, content} ->
+        ArticlePort.wikipedia_to_article(content)
+
+      {:error, reason} ->
+        if attempts > 0 do
+          ConcurrencyManager.adjust_concurrency(:decrease)
+          :timer.sleep(500)
```

Comment:
I'll implement a scale on this `:timer.sleep()` with a `fetch_wikipedia_page/3`, where the 3rd param is a `ms` that will be calculated.